pkg/sidecar/appclone_test.go

```diff
@@ -0,0 +1,226 @@
+package sidecar
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+var _ = Describe("Test RunCloneCommand cloning logic", func() {
+
+	var (
+		cfg               *Config
+		fakeBackupFile    string // as named in fakeServer.validXBStream
+		fakeMasterServer  *fakeServer
+		fakeReplicaServer *fakeServer
+		// Normally, these are true k8s services, each listening on
+		// SidecarServerPort. Since we can't simulate that in unit tests, we put
+		// each "service" on its own port.
+		masterServiceAddr          = fmt.Sprintf(":%d", serverPort)
```

Comment:
The fake master service runs on `8080` and the fake replica servers on `8081`. I'm happy to change these or use a different approach, if you think there's a better way.

Reply:
Ah, thank you @calind, I did not know about that function, very handy! Updated here: e3bb4a0ca323b290c270212d5d314792bc492c77